worker/provisioner/provisioner_test.go

```diff
@@ -1380,10 +1473,241 @@ func (s *ProvisionerSuite) TestProvisionerObservesMachineJobs(c *gc.C) {
 	}
 }
 
+func assertAvailabilityZoneMachines(c *gc.C,
+	machines []*state.Machine,
+	obtained []*provisioner.AvailabilityZoneMachine,
+	expected []*provisioner.AvailabilityZoneMachine,
+) {
+	sort.Sort(provisioner.ByPopulationThenName(expected))
+	sort.Sort(provisioner.ByPopulationThenName(obtained))
+
+	if len(machines) > 0 {
+		for _, m := range machines {
+			zone, err := m.AvailabilityZone()
+			c.Assert(err, jc.ErrorIsNil)
+			for _, zoneInfo := range obtained {
+				if zone == zoneInfo.ZoneName {
+					c.Assert(zoneInfo.MachineIds.Contains(zone), gc.Equals, true)
+
+				}
+			}
+		}
+	}
+}
+
+func (s *ProvisionerSuite) TestAvailabilityZoneMachinesStartMachines(c *gc.C) {
+	// Per provider dummy, there will be 3 available availability zones.
+	task := s.newProvisionerTask(c, config.HarvestDestroyed, s.Environ, s.provisioner, &mockDistributionGroupFinder{}, mockToolsFinder{})
+	defer stop(c, task)
+
+	machines := make([]*state.Machine, 4)
+	for i, _ := range machines {
+		m, err := s.addMachine()
+		c.Assert(err, jc.ErrorIsNil)
+		s.checkStartInstance(c, m)
```

Comment:
Perhaps we need a checkStartInstances(c, []machine)
We call checkStartInstances() after the loop.

Reply:
done, addMachines() is new to the tests as well.